os_info/src/windows/winapi.rs

```diff
@@ -114,17 +122,80 @@ fn version_info() -> Option<OSVERSIONINFOEX> {
     }
 }
 
+fn str_to_wide(value: &str) -> Vec<WCHAR> {
```

Comment:
I suppose that using __A functions family (for example `RegOpenKeyExA` instead of `RegOpenKeyExW`) will require a little less code.

Reply:
Actually, it's either a wash or a bit more.

The strings passed into `RegOpenKeyExA` and `RegQueryValueExA` cannot be used as-is; a null terminator has to be included...
```Rust
"ReleaseId\0"
"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\0"
```

But that's a minor problem.  The big kick is `RegQueryValueExA`.  The string returned from `RegQueryValueExA` will have to be converted from ANSI to UTF-8 using the current code page.  That adds a few more API calls and more buffer management.

And, it's my understanding that ANSI to UTF-8 is not available so the ANSI string will have to be converted to UTF-16 (widestring) then to UTF-8.  If that's true then we may as well just get the string as UTF-16.